msteams-platform/bots/how-to/format-ai-bot-messages.md

```diff
@@ -0,0 +1,488 @@
+
+Citations to your bot message include:
+
+* [In-text citations](#add-in-text-citations) to cite your text using the [X] format at any point within the text.
+* [Details to citation reference](#add-details-to-citation-reference) to list the corresponding references of the in-text citations.
+
+### Add in-text citations
+
+In-text citations appear as numbers, each corresponding to a reference that can include the title, keywords, an abstract (or excerpt), hyperlink, and sensitivity information. A citation can be inserted anywhere within the text.
+
+:::image type="content" source="../../assets/images/bots/ai-bot-inline-citation.png" border="false" alt-text="Screenshot shows an AI bot response with in-text citation.":::
+
+The following code snippet shows the format of in-text citations to be added to your bot's message:
+
+```javascript
+await context.sendActivity({
+    type: ActivityTypes.Message,
+    text: 'Hey I'm a friendly AI bot. This message is generated via AI - $(txt) [1]', // cite with [1]
+});
+```
+
+### Add details to citation reference
+
+References for these in-text citations appear as pop-ups. The pop-up provides key details such as the title of the citation, the link to the source, and a relevant quote from the source.
+
+For a bot built using Teams AI library, Teams displays the in-line citations when the message includes `citation` array in the `entities` object.
+
+The following code snippet shows the citation reference added to your bot's message:
+
+```javascript
+ await context.sendActivity({
+         type: ActivityTypes.Message,
+        text: `Hey I'm a friendly AI bot. This message is generated via AI - $(txt) [1]`, // cite with [1]
+        entities: [
+          {
+            type: "https://schema.org/Message",
+            "@type": "Message",
+            "@context": "https://schema.org",
+            "@id": "",
+            citation: [
+              {
+                "@type": "Claim",
+                position: 1, // Required. Should match the [1] in the text above
+                appearance: {
+                  "@type": "DigitalDocument",
+                  name: "Some secret citation", // Title
+                  url: "https://example.com/claim-1", // Hyperlink on the title
+                  abstract: "Excerpt", // Excerpt (abstract)
+                  keywords: ["Keyword1 - 1", "Keyword1 - 2", "Keyword1 - 3"], // Keywords
+                  usageInfo: {
+                    "@type": "CreativeWork",
+                    name: "Confidential \\ Contoso FTE", // Sensitivity title
+                    description: "Only accessible to Contoso FTE", // Sensitivity description
+                  },
+                },
+              },
+            ],
+          },
+        ],
+}
+```
+
```

Comment:
As discussed with Yoga, bot developer adding some citations in entities fields isn't included right now because we disabled the dropdown. The developer can provide it just won't show up anywhere.

Yoga to follow up on the citation limits.

Reply:
@yogapat to confirm the citation limit.

Reply:
@yogapat, can you confirm if the "bot developer can provide some citations in the entities field which are not referenced from the text of the message." is included in the citation limit?